registry/import.go

```diff
@@ -0,0 +1,153 @@
+package registry
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"encoding/json"
+	"fmt"
+	"github.com/cozy/cozy-apps-registry/asset"
+	"github.com/cozy/cozy-apps-registry/config"
+	"github.com/go-kivik/kivik"
+	"io"
+	"io/ioutil"
+	"path"
+	"strings"
+)
+
+func cleanCouch() error {
+	fmt.Printf("Clean CouchDB\n")
+	for _, db := range couchDatabases() {
+		name := db.Name()
+		fmt.Printf("  Clean CouchDB %s\n", name)
+		if err := client.DestroyDB(ctx, name); err != nil {
+			return err
+		}
+		if err := client.CreateDB(ctx, name); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func importCouchDocument(reader io.Reader, db string, id string) (string, error) {
+	fmt.Printf("Import CouchDB document %s.%s\n", db, id)
+
+	var doc json.RawMessage
+	if err := json.NewDecoder(reader).Decode(&doc); err != nil {
+		return "", err
+	}
+
+	c := client.DB(ctx, db)
+	return c.Put(ctx, id, doc)
```

Comment:
It's not very slow to import the documents one by one in Couch instead of using [bulk](http://docs.couchdb.org/en/stable/api/database/bulk-api.html#db-bulk-docs)?

Reply:
We can't use bulk. In tar, doc is followed by it attachments, and there is too many documents now to expect storing it in memory between…
We have to read doc, then attachment, then the next doc, etc.

Reply:
so think out of the box. Why don't you untar on disk in a temporary space before bulk importing all docs, then adding attachements if any, then remove temporary space ?
writing on disk is probably faster than writing each doc one by one on couch over the network (without keepalive I suppose)

Reply:
I don't see the issue: you don't need to keep all the documents in memory. You can keep a slice of documents, and its length reaches 100, you can flush (ie sending a bulk request with the 100 docs, and emptying the slice).

Reply:
You need to have the doc created when importing it attachments…
So you also need to store attachments, and so not only flush each X docs, but also exactly between 2 documents + attachments.
And no way to go back on the tar, so the overall process will be more complicated…

Reply:
Why are you not a le to make it with 2 passes, firdt one for the docs, second one for the attachements ?
why haven't you organise your export tar with distinct objets (such as directories) for docs, attachements and swift objets ?

Reply:
It's organized by directory (couchdb/id.json, couchdb/id/attachment, swift/such/a/path), but tar reader go API has only a single `Next` command to iterate over the archive… :fearful: 
And you need to read all a document (header + content) to be able to access the next one… :fearful:

Reply:
ok, so read it to /dev/null and start over reading tar from the beginning to grab attachments.
or simply generate your tar at export with all docs first, then attachments.

Reply:
or simply don't use go tar reader.